locast2dvr/dvr.py

```diff
@@ -176,6 +176,10 @@ def get_stations(self) -> list:
                 self.station_service_mapping[str(
                     station['id'])] = d.locast_service
 
+        if self.config.remap_by_name:
+              stations = _remap_by_name(stations)
+              self.config.remap_by_name = False
```

Comment:
I noticed though that it seemed to cause problems for Plex without it.  I think it was remapping every time I went into the guide, settings, etc.  I noticed that if I went to the Live TV/DVR settings in Plex, that when the page came up, it would have 0 channels and would switch to the right count after a second or less.  Also, I was getting quite a few recording errors - maybe due to the extra overhead that this process was doing?

As you are probably aware, the stations list of dictionaries can get quite huge since it has the EPG data in it too.  Maybe there'd be a way to split that up into two lists, one with just the channel info and one with the listings info, both sharing a unique value - perhaps the "id" entry that's in the current dictionaries.  The stations one would have all of the dictionary items instead of "listings" array, and the listings one would have the "id" and the "listings" array.  You could tweak/remap/whatever the stations one with a much smaller structure, and it seems to make sense that it would perform better.  And you'd only have to update the listings one whenever you were updating the EPG data.

Reply:
> I noticed though that it seemed to cause problems for Plex without it. I think it was remapping every time I went into the guide, settings, etc. I noticed that if I went to the Live TV/DVR settings in Plex, that when the page came up, it would have 0 channels and would switch to the right count after a second or less. Also, I was getting quite a few recording errors - maybe due to the extra overhead that this process was doing?

I honestly doubt that. If you run with `-v`, you should see the HTTP call that Plex makes and it should not request `epg.xml` every single time. And even if it would, processing in locast2dvr is actually quite fast, compared to the work that Plex (or Emby) need to do to parse the xml and create an actual guide out of it.

> As you are probably aware, the stations list of dictionaries can get quite huge since it has the EPG data in it too. Maybe there'd be a way to split that up into two lists, one with just the channel info and one with the listings info, both sharing a unique value - perhaps the "id" entry that's in the current dictionaries. The stations one would have all of the dictionary items instead of "listings" array, and the listings one would have the "id" and the "listings" array. You could tweak/remap/whatever the stations one with a much smaller structure, and it seems to make sense that it would perform better. And you'd only have to update the listings one whenever you were updating the EPG data.

I thought about this, but since all of this still happens in matter of milliseconds, I decided not to prematurely optimize.. You can see that a call to `get_stations()` is still very fast when you request `/tuner.m3u`. It pretty much instantly responds, even though the same data is loaded as for `/epg.xml`.

Reply:
Or, maybe we create two separate functions.  One that runs when processing data from a new/updated load from Locast, and another for general processing, that doesn't do any further manipulation of the data.

Reply:
> > I noticed though that it seemed to cause problems for Plex without it. I think it was remapping every time I went into the guide, settings, etc. I noticed that if I went to the Live TV/DVR settings in Plex, that when the page came up, it would have 0 channels and would switch to the right count after a second or less. Also, I was getting quite a few recording errors - maybe due to the extra overhead that this process was doing?
> 
> I honestly doubt that. If you run with `-v`, you should see the HTTP call that Plex makes and it should not request `epg.xml` every single time. And even if it would, processing in locast2dvr is actually quite fast, compared to the work that Plex (or Emby) need to do to parse the xml and create an actual guide out of it.
> 
I've noticed though that if I run without the remap by name, that delay seems to go away, that's why I wondered if it might be interfering.  Who knows though, maybe something else was going on with the server it's running on.